db/forward_iterator.cc

```diff
@@ -369,7 +369,70 @@ void ForwardIterator::Seek(const Slice& internal_key) {
   } else if (immutable_status_.IsIncomplete()) {
     ResetIncompleteIterators();
   }
+
   SeekInternal(internal_key, false);
+  if (read_options_.async_io) {
+    SeekInternalAfterAsyncIO(internal_key);
+  }
+}
+
+void ForwardIterator::SeekInternalAfterAsyncIO(const Slice& internal_key) {
```

Comment:
Can we try to just call SeekInternal() again, instead of duplicating most of the logic here? Maybe SeekInternal() can return something to indicate that seek is not complete yet. We can do some optimizations there for the 2nd seek, like not calling FindFileInRange() and SetFileIndex() for the level iterator.